atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -217,6 +217,9 @@
     @Idempotent
     void commit(TransactionService transactionService) throws TransactionFailedException;
 
+    @Idempotent
+    void commit(CommitRequest commitRequest) throws TransactionFailedException;
```

Comment:
this won't be the right interface, because users almost never call commit directly themselves - the transaction manager does it. If you want Transaction to be aware of this, I think it gets to be aware of the lock watches that are registered and accept the read ranges in a method.